tests/ClientTest.php

```diff
@@ -469,11 +473,76 @@ public function testThrottlingGlobalShortLimitReached()
         $clientMock->getAttributes();
     }
 
+    /**
+     * @return array
+     */
+    public function providerGetProductVariationsSetsWithParameter()
+    {
+        return [
+            'With parameter is an empty array' => [
+                [],
+                []
+            ],
+            'With parameter is not an empty array' => [
+                [
+                    'variationCategories',
+                    'variationSalesPrices',
+                    'variationAttributeValues',
+                    'variationProperties',
+                    'properties',
+                    'units'
+                ],
+                [
+                    'variationCategories',
+                    'variationSalesPrices',
+                    'variationAttributeValues',
+                    'variationProperties',
+                    'properties',
+                    'units'
+                ]
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider providerGetProductVariationsSetsWithParameter
+     *
+     * @param array $with
+     * @param array $expectedWith
+     * @throws \ReflectionException
+     */
+    public function testGetProductVariationsSetsWithParameter(array $with, array $expectedWith)
+    {
+        $debugMock = $this->getMockBuilder(Debugger::class)->disableOriginalConstructor()->getMock();
+        $logMock = $this->getMockBuilder(Logger::class)->disableOriginalConstructor()->getMock();
+        $configMock = $this->getMockBuilder(PlentyConfig::class)->setMethods(['getDomain'])->getMock();
+
+        $clientMock = $this->getMockBuilder(Client::class)
+            ->setConstructorArgs([$configMock, $logMock, $logMock, $debugMock])
+            ->setMethods(['getEndpoint', 'call'])
+            ->getMock();
+
+        $response = $this->getResponseMock('{}', 200);
+
+        $clientMock->expects($this->once())->method('getEndpoint')->with(
+            'items/variations',
+            [
+                'with' => $expectedWith,
+                'isActive' => true,
+                'itemId' => ['1']
+            ]
+        );
+
+        $clientMock->expects($this->any())->method('call')->willReturn($response);
+
+        $clientMock->getProductVariations(['1'], $with);
+    }
+
     /* ------ helper functions ------ */
 
     /**
      * @param $methods
-     * @return \Findologic\Plentymarkets\Client|MockObject
+     * @return Client|MockObject
      * @throws \ReflectionException
      */
     protected function getClientMock($methods)
```

Comment:
Additionally we can typehint the return value to be `Client`, because `MockObject` is always an instance of the main-class that it was created from.